### ISSUE TITLE
Disappearing pilots in heats after populating it.

### ISSUE THREAD

**Issue by evilmerlin** [RotorHazard/RotorHazard#823]
**Describe the bug**
When selecting pilots via the drop down menu in the heats, after selecting and going to some other page, the initially selected pilots would be gone. Leaving a blank seat in the heat and I have to reselect them again. However, not all selected pilots are gone. I created 4 heats and populated it with 4 pilots each. Some heats would end up having only 1 pilot and some with 2. 

It is however quite intermittent. I have seen it happen twice during the entire event and don't see a pattern in the replication.

**To Reproduce**
1. Create new class
2. Add heats
3. Fill heats
4. Go to other page or other browser tabs
5. Go back to Format and check heats

**Expected behavior**
For the selected pilots in the heats to stay where they are and not disappear.

[rh_logs_20231210_153654.zip](https://github.com/RotorHazard/RotorHazard/files/13649948/rh_logs_20231210_153654.zip)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome
 - Version: 119.0.6045.200

**Smartphone (please complete the following information):**
No Smartphone used

**Comment by evilmerlin** [comment]
I am unable to consistently reproduce it but it was observed. 
Did not check to see if the error is present when plugin is not used but the plugin is upload only and does not communicate back to the RH server.

So in a nutshell, I had just purged the database and was manually recreating the race heats for the elimination rounds. Created a new class, added the heats and started to populate them with pilots. Once done I went off to do other stuff (open new browser tabs and checked spreadsheets etc etc), when I went back to RH to start the elimination rounds did I see that the some of the pilots I had entered had disappeared and needed to be re-entered.
